components/job-orchestration/job_orchestration/scheduler/compress/compression_scheduler.py

```diff
@@ -146,9 +156,43 @@ def _process_s3_input(
         paths_to_compress_buffer.add_file(object_metadata)
 
 
-def search_and_schedule_new_tasks(db_conn, db_cursor, clp_metadata_db_connection_config):
+def _register_dataset(
+    db_conn,
+    db_cursor,
+    table_prefix: str,
+    dataset_name: str,
+    clp_storage_type: StorageType,
+    archive_storage_directory: Path,
+    existing_datasets: Set[str],
+) -> None:
+    add_dataset(
+        db_cursor,
+        table_prefix,
+        dataset_name,
+        clp_storage_type,
+        archive_storage_directory,
+    )
+    create_metadata_db_tables(db_cursor, table_prefix, dataset_name)
+    db_conn.commit()
```

Comment:
Can we move these statements into `add_dataset`? I can't think of a good reason to leave them separate for now.